### PR TITLE
Add enrich policy DELETE API

### PR DESCRIPTION
**Pull request by hub-cap**
This commit wires up the Rest calls and Transport calls for DELETE enrich
policy, as well as tests and rest spec additions.

### CODE REVIEW

**Comment by hub-cap** [comment]
@elasticmachine run elasticsearch-ci/1

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/enrich/action/DeleteEnrichPolicyAction.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class DeleteEnrichPolicyAction extends Action<AcknowledgedResponse> {
+
+    public static final DeleteEnrichPolicyAction INSTANCE = new DeleteEnrichPolicyAction();
+    public static final String NAME = "cluster:admin/xpack/enrich/delete";
+
+    protected DeleteEnrichPolicyAction() {
```
Comment (martijnvg) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/enrich/action/DeleteEnrichPolicyAction.java]: I think this constructor can be private?

File: x-pack/plugin/src/test/resources/rest-api-spec/test/enrich/10_basic.yml [hunk 2]
```diff
@@ -10,5 +10,9 @@
           enrich_key: baz
           enrich_values: ["a", "b"]
           schedule: "*/120"
+  - is_true: acknowledged
 
+  - do:
+      enrich.delete_policy:
+        name: policy-crud
```
Comment (martijnvg) [inline x-pack/plugin/src/test/resources/rest-api-spec/test/enrich/10_basic.yml]: maybe also do a get policy after the delete to ensure a 404 is returned?

**Review by martijnvg — APPROVED** [review]
I left 2 small comments, LGTM otherwise

**Review by jakelandis — APPROVED** [review]

**Pull request merged**